packages/svelte/src/internal/client/dom/blocks/branches.js

```diff
@@ -0,0 +1,188 @@
+/** @import { Effect, TemplateNode } from '#client' */
+import { is_runes } from '../../context.js';
+import { Batch, current_batch } from '../../reactivity/batch.js';
+import {
+	branch,
+	destroy_effect,
+	move_effect,
+	pause_effect,
+	resume_effect
+} from '../../reactivity/effects.js';
+import { set_should_intro, should_intro } from '../../render.js';
+import { hydrate_node, hydrating } from '../hydration.js';
+import { create_text, should_defer_append } from '../operations.js';
+
+/**
+ * @typedef {{ effect: Effect, fragment: DocumentFragment }} Branch
+ */
+
+/**
+ * @template Key
+ */
+export class BranchManager {
+	/** @type {TemplateNode} */
+	anchor;
+
+	/** @type {Map<Batch, Key>} */
+	#batches = new Map();
+
+	/** @type {Map<Key, Effect>} */
+	#onscreen = new Map();
+
+	/** @type {Map<Key, Branch>} */
+	#offscreen = new Map();
+
+	#legacy = !is_runes();
+
+	/**
+	 * Whether to pause (i.e. outro) on change, or destroy immediately.
+	 * This is necessary for `<svelte:element>`
+	 */
+	#transition = true;
+
+	/**
+	 * @param {TemplateNode} anchor
+	 * @param {boolean} transition
+	 */
+	constructor(anchor, transition = true) {
+		this.anchor = anchor;
+		this.#transition = transition;
+	}
+
+	#commit = () => {
+		var batch = /** @type {Batch} */ (current_batch);
+
+		// if this batch was made obsolete, bail
+		if (!this.#batches.has(batch)) return;
+
+		var key = /** @type {Key} */ (this.#batches.get(batch));
+
+		var onscreen = this.#onscreen.get(key);
+
+		if (onscreen) {
+			// effect is already in the DOM — abort any current outro
+			resume_effect(onscreen);
+		} else {
+			// effect is currently offscreen. put it in the DOM
+			var offscreen = this.#offscreen.get(key);
+
+			if (offscreen) {
+				this.#onscreen.set(key, offscreen.effect);
+				this.#offscreen.delete(key);
+
+				// remove the anchor...
+				/** @type {TemplateNode} */ (offscreen.fragment.lastChild).remove();
+
+				// ...and append the fragment
+				this.anchor.before(offscreen.fragment);
+				onscreen = offscreen.effect;
+			}
+		}
+
+		this.#batches.delete(batch);
+
+		for (const [b, k] of this.#batches) {
+			if (b === batch) break;
+
+			const offscreen = this.#offscreen.get(k);
+
+			if (offscreen) {
+				destroy_effect(offscreen.effect);
+				this.#offscreen.delete(k);
+			}
+
+			this.#batches.delete(b);
+		}
+
+		// outro/destroy effects
+		for (const [k, effect] of this.#onscreen) {
+			if (k === key) continue;
```

Comment:
Why only continue and not break? Doesn't that mean effects of newer batches that haven't committed yet also run?